src/test/webServer.ts

```diff
@@ -54,12 +54,11 @@ class LaunchServer {
     let processExitedReject = (error: Error) => { };
     this._processExitedPromise = new Promise((_, reject) => processExitedReject = reject);
 
-    if (this.config.waitForPort) {
-      const portIsUsed = !await canBindPort(this.config.waitForPort);
-      if (portIsUsed && this.config.strict)
-        throw new Error(`Port ${this.config.waitForPort} is used, make sure that nothing is running on the port or set strict:false in config.launch.`);
-      if (portIsUsed)
+    const portIsUsed = !await canBindPort(this.config.port);
+    if (portIsUsed) {
+      if (this.config.reuseExistingServer)
         return;
+      throw new Error(`Port ${this.config.port} is used, make sure that nothing is running on the port or set strict:false in config.launch.`);
     }
 
     console.log(`Launching '${this.config.command}'...`);
```

Comment:
Let's not log until somebody complains. I don't want this polluting my logs every time I run my tests.

```suggestion
```